de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/CustomTreeCell.java

```diff
@@ -163,7 +153,21 @@ public CustomTreeCell(WorkspaceFileSystem workspaceFileSystem, FileImageProvider
 			ArrayList<java.io.File> sources = new ArrayList<java.io.File>();
 
 			for (FileTreeElement entry : services.rcpSelectionService.getCurrentSelectionsFromExplorer()) {
-				sources.add(new java.io.File(entry.getAbsolutePath()));
+				boolean found = false;
+				String dirName = entry.getAbsolutePath();
+				// if a matching subpath is found (e.g. /something/dir exists and a new file
+				// /something/dir/file.file is added) the search ends and the path won't be
+				// added to the content list of files
+				sourcesContainLoop: for (File file : sources) {
+					if (dirName.contains(file.getAbsolutePath())) {

```

Comment:
As the source is always in the project we should only compare relative pathes

Reply:
java io File does not provide a relative path sadly